sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -760,7 +762,14 @@ private[hive] case class InsertIntoHiveTable(
   override def children: Seq[LogicalPlan] = child :: Nil
   override def output: Seq[Attribute] = child.output
 
-  override lazy val resolved: Boolean = childrenResolved && child.output.zip(table.output).forall {
+  val numDynamicPartitions = partition.values.count(_.isEmpty)
+  val tableOutput = {
+    val hiveTable = table.asInstanceOf[MetastoreRelation]
```

Comment:
This cast is worrisome.  Can we instead make `table` a `MetastoreRelation` so this is enforced when we create this operator?